File: app/src/main/java/com/jppedrosa/mvpskeleton/MainPresenter.java
package com.jppedrosa.mvpskeleton;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.jppedrosa.mvpskeleton.core.BasePresenter;

/**
 * @author João Pedro Pedrosa (<a href="mailto:joaopopedrosa@gmail.com">joaopopedrosa@gmail.com</a>) on 07/09/2022.
 */
public class MainPresenter extends BasePresenter<MainView> {

    protected MainPresenter(@NonNull MainView view,
                            @NonNull Context mContext) {
        super(view, mContext);
    }
}

File: app/src/main/java/com/jppedrosa/mvpskeleton/ui/MainPresenter.kt
package com.jppedrosa.mvpskeleton.ui

import android.content.Context
import com.jppedrosa.mvpskeleton.core.BasePresenter

/**
 * @author João Pedro Pedrosa ([joaopopedrosa@gmail.com](mailto:joaopopedrosa@gmail.com)) on 07/09/2022.
 */
class MainPresenter(
    view: MainView,
    mContext: Context
) : BasePresenter<MainView>(view, mContext) {

    fun onStart() {

    }

    fun onResume() {
        this.view?.startInternetConnectionListener()
    }

    fun onDestroy() {
        this.view?.stopInternetConnectionListener()
    }
}